main.cpp

```diff
@@ -19,13 +19,32 @@ using namespace clang::tooling;
 
 class CastCallBack : public MatchFinder::MatchCallback {
 public:
-    CastCallBack(Rewriter& rewriter) {
-        // Your code goes here
-    };
+	CastCallBack(Rewriter& _rewriter) : rewriter_(_rewriter) {};
 
-    void run(const MatchFinder::MatchResult &Result) override {
-        // Your code goes here
-    }
+	void run(const MatchFinder::MatchResult& Result) override {
+		if (auto* cast = Result.Nodes.getNodeAs<CStyleCastExpr>("cast")) {
+			auto type = Lexer::getSourceText(

```

Comment:
Please describe what the `getSourceText()` function does.

Reply:
`getSourceText()` returns a _string_ for the source in the specified range. In my realisation this function is used to get the type.